botorch/models/utils/inducing_point_allocators.py

```diff
@@ -0,0 +1,388 @@
+        u = (mean - best_f) / sigma if self._maximize else -(mean - best_f) / sigma
+        return sigma * (phi(u) + u * Phi(u))
+
+
+class GreedyVarianceReduction(InducingPointAllocator):
+    r"""
+    The inducing point allocator proposed by [burt2020svgp]_, that
+    greedily chooses inducing point locations with maximal (conditional)
+    predictive variance.
+    """
+
+    def allocate_inducing_points(
+        self,
+        inputs: Tensor,
+        covar_module: Module,
+        num_inducing: int,
+        input_batch_shape: torch.Size,
+    ) -> Tensor:
+        """
+        Greedily initialize `num_inducing` inducing points following [burt2020svgp]_.
+
+        Args:
+            inputs: A (*batch_shape, n, d)-dim input data tensor.
+            covar_module: GPyTorch Module returning a LinearOperator kernel matrix.
+            num_inducing: The maximun number (m) of inducing points (m <= n).
+            input_batch_shape: The non-task-related batch shape.
+
+        Returns:
+            A (*batch_shape, m, d)-dim tensor of inducing point locations.
+        """
+
+        return self._allocate_inducing_points(
+            inputs, covar_module, num_inducing, input_batch_shape, UnitQualityFunction()
+        )
+
+
+class GreedyImprovementReduction(InducingPointAllocator):
+    r"""
+    An inducing point allocator that greedily chooses inducing points with large
+    predictive variance and that are in promising regions of the search
+    space (according to the model form the previous BO step), see :cite:`moss2023IPA`.
+    """
+
+    def __init__(self, model: Model, maximize: bool):
+        r"""
+
+        Args:
+            model: The model fitted during the previous BO step.
+            maximize: Set True if we are performing function maximization, else
+                set False.
+        """
+        self._model = model
+        self._maximize = maximize
+
+    def allocate_inducing_points(
+        self,
+        inputs: Tensor,
+        covar_module: Module,
+        num_inducing: int,
+        input_batch_shape: torch.Size,
+    ) -> Tensor:
+        """
+        Greedily initialize the `num_inducing` inducing points following the IMP-DPP
+        strategy of [moss2023ipa]_.
+
+        Args:
+            inputs: A (*batch_shape, n, d)-dim input data tensor.
+            covar_module: GPyTorch Module returning a LinearOperator kernel matrix.
+            num_inducing: The maximun number (m) of inducing points (m <= n).
+            input_batch_shape: The non-task-related batch shape.
+
+        Returns:
+            A (*batch_shape, m, d)-dim tensor of inducing point locations.
+        """
+
+        return self._allocate_inducing_points(
+            inputs,
+            covar_module,
+            num_inducing,
+            input_batch_shape,
+            ExpectedImprovementQualityFunction(self._model, self._maximize),
+        )
+
+
+def _pivoted_cholesky_init(
```

Comment:
This is essentially a clone of https://github.com/pytorch/botorch/blob/d026e805f34bcffc9003720f1822e06f92ec08ab/botorch/models/approximate_gp.py#L493. Would you mind
1. changing the function here so that `quality_scores` is an optional arg (in which case they effectively would be a unit vector but we don't really need to do that in the implementation)
2. Modifying the original function to just call this one while raising a `DeprecationWarning` pointing to this function instead?

FYI: @j-wilson has implemented some pivoted cholesky functionality as well (e.g. here https://github.com/pytorch/botorch/blob/d026e805f34bcffc9003720f1822e06f92ec08ab/botorch/utils/probability/linalg.py#L74) - it may be worth thinking how that could be used here (though I'm not suggesting to make this part of this PR, that should stay separate).

Reply:
So I actually move the old _pivoted_cholesky_init from approximate_gp.py into this file and deleted the old code. This new one is used for both the old (greedy variance reduction) and the new IPAs.

Do you want me to add the old one back into approximate_gp.py?

Reply:
Oh, I must have missed that somehow, nvm.